routemaster/services/throttle.rb

```diff
@@ -0,0 +1,93 @@
+require 'routemaster/services'
+require 'routemaster/mixins/log'
+
+# throttle |ˈθrɒt(ə)l|
+#   noun
+#   a device controlling the flow of fuel or power to
+#   an engine: the engines were at full throttle.
+#
+module Routemaster
+  module Services
+    class Throttle
+      MAX_HP = 100
+
+      class EarlyThrottle < StandardError
+        attr_reader :message
+        def initialize(subcriber_name)
+          @message = "Throttling batch deliveries to the '#{subcriber_name}' subscriber."
+        end
+      end
+
+
+      def initialize(batch: nil, subscriber: nil)
```

Comment:
Looks like you no longer need the ability to pass a batch here — which makes sense, as the throttling is per-subscriber only.